UIElements/triangularCalibration.py

```diff
@@ -25,104 +28,177 @@ def cutTestPaternTriangular(self):
+        self.enterValuesT.disabled = True
+
+        print "Iterating for machine parameters"
+
+        # Iterate until error tolerance is achieved or maximum number of iterations occurs
+
+        while (((abs(ChainErrorCut1) >= acceptableTolerance ) or (abs(ChainErrorCut2) >= acceptableTolerance)) and (n < numberOfIterations)):
+            n += 1
+
+            # Calculate chain lengths for current estimated parameters
+
+            MotorDistanceCut1Est = math.sqrt(math.pow(motorXcoord,2)+math.pow(motorYcoordEst,2))
+            MotorDistanceCut2Est = math.sqrt(math.pow(motorXcoord,2)+math.pow(motorYcoordEst + distBetweenCuts,2))
+
+            ChainLengthCut1Est = (motorSprocketRadius * (3.14159 - math.acos(motorSprocketRadius / MotorDistanceCut1Est) - math.acos((motorYcoordEst) / MotorDistanceCut1Est))) + math.sqrt(math.pow(MotorDistanceCut1Est,2) - math.pow(motorSprocketRadius,2)) - rotationRadiusEst
+            ChainLengthCut2Est = (motorSprocketRadius * (3.14159 - math.acos(motorSprocketRadius / MotorDistanceCut2Est) - math.acos((motorYcoordEst + distBetweenCuts) / MotorDistanceCut2Est))) + math.sqrt(math.pow(MotorDistanceCut2Est,2) - math.pow(motorSprocketRadius,2)) - rotationRadiusEst
+
+            # Determine chain length errors for current estimated machine parameters versus the measured parameters
+
+            ChainErrorCut1 = ChainLengthCut1Est - ChainLengthCut1
+            ChainErrorCut2 = ChainLengthCut2Est - ChainLengthCut2
+            ChainErrorCutRatio = ChainErrorCut1 / ChainErrorCut2
+
+            # Establish a correction value
+
+            Correction = (ChainErrorCut1 + ChainErrorCut2) / 2
+
+            # Develop a printable motor Y offset value to update the user
+
+            motorYoffsetEstPrint = motorYcoordEst - distWorkareaTopToCut - (bitDiameter / 2)
+
+            print "N: " + str(n) + ", Motor Spacing: " + str(round(motorSpacing, 3)) + ", Motor Y Offset: " + str(round(motorYoffsetEstPrint, 3)) + ", Rotation Disk Radius: " + str(round(rotationRadiusEst, 3))
+
+            # Update the machine parameters based on the current chain length errors
+
+            if (ChainErrorCutRatio > 0):
+                motorYcoordEst -= Correction * motorYcoordCorrectionScale
+            else:
+                rotationRadiusEst -= Correction * rotationRadiusCorrectionScale
+
+            # If we get unrealistic values, reset and try again with smaller steps
+
+            if (motorYcoordEst < -(workspaceHeight/4) or rotationRadiusEst < -100):
+                    motorYcoordEst = distWorkareaTopToCut + (bitDiameter / 2) + (workspaceHeight / 4)
+                    rotationRadiusEst = 100
+                    motorYcoordCorrectionScale = float(motorYcoordCorrectionScale/2)
+                    rotationRadiusCorrectionScale = float(rotationRadiusCorrectionScale/2)
+                    print "Estimated values out of range, trying again with smaller steps"
+
+        if n == numberOfIterations:
```

Comment:
at least as a debug option, we should output both the number of iterations, and the accuracy we think we are at.